src/librustc_target/spec/mod.rs

```diff
@@ -1176,8 +1184,10 @@ impl Target {
         match *target_triple {
             TargetTriple::TargetTriple(ref target_triple) => {
                 // check if triple is in list of supported targets
-                if let Ok(t) = load_specific(target_triple) {
-                    return Ok(t)
+                match load_specific(target_triple) {
+                    Ok(t) => return Ok(t);
+                    LoadTargetError::BuiltinTargetNotFound(_) => (),
+                    LoadTargetError::Other(e) => return Err(e),
```

Comment:
I'm not so sure about these load errors, it seems to me that they are a compiler bug. Either we panic on them and test them, or we allow the user to work around them by specifying a json file. I'd prefer the former.

Reply:
> I'm not so sure about these load errors, it seems to me that they are a compiler bug.

When you target, say, an `ios` target, creating the Target executes code like this: https://github.com/rust-lang/rust/blob/master/src/librustc_target/spec/apple_ios_base.rs#L29

Which can fail, say on Linux, if that tool is not available. 

So in that case, load target failed, for a reason that AFAICT isn't a compiler bug, but an user bug: its running the compiler without the appropriate tools in the path.

Reply:
I don't know if reporting those errors breaks somebody's workflow. Where they are relying, e.g., on the builtin target failing for external reasons, to override it with a json file. Honestly, that use case is better addressed with a `--target-file=file.json` (or just `--target=file.json`) CLI option that explicitly requests obtaining the target from a json file.

Reply:
that makes sense